homeassistant/components/alexa/state_report.py

```diff
@@ -154,14 +159,19 @@ async def async_send_changereport_message(
 
     response_json = json.loads(response_text)
 
-    if (
-        response_json["payload"]["code"] == "INVALID_ACCESS_TOKEN_EXCEPTION"
-        and not invalidate_access_token
-    ):
-        config.async_invalidate_access_token()
-        return await async_send_changereport_message(
-            hass, config, alexa_entity, alexa_properties, invalidate_access_token=False
-        )
+    if response_json["payload"]["code"] == "INVALID_ACCESS_TOKEN_EXCEPTION":
+        if invalidate_access_token:
```

Comment:
Note that this fixes a bug; the logic was wrong.
This could be moved to a separate PR.